lifetimes/estimation.py

```diff
@@ -36,6 +37,185 @@ def _print_params(self):
             s += "%s: %.2f, " % (p, value)
         return s.strip(', ')
 
+class BetaGeoBetaBinomFitter(BaseFitter):
+    """
+    Also known as the Beta-Geometric/Beta-Binomial Model Model [1].
+
+    Future purchases opportunities are treated as discrete points in time. In the literature,
+    the model provides a better fit than the Pareto/NBD model for a nonprofit organization
+    with regular giving patterns.
+
+    The model is estimated with a recency-frequency matrix with n transaction opportunities.
+
+    [1] Fader, Peter S., Bruce G.S. Hardie, and Jen Shang (2010),
+        "Customer-Base Analysis in a Discrete-Time Noncontractual Setting,"
+        Marketing Science, 29 (6), 1086-1108.
+
+    """
+    
+    def __init__(self, penalizer_coef=0.):
+        self.penalizer_coef = penalizer_coef
+
+    @staticmethod
+    def _loglikelihood(params, x, tx, T):
+
+        N = len(x)
```

Comment:
to avoid ambiguity, can you use `x.shape[0]`? (I think it is safe to assume only numpy arrays will be allowed in here.